TPP.Core/Commands/Definitions/BadgeCommands.cs

```diff
@@ -134,15 +141,150 @@ public async Task<CommandResult> SelectBadge(CommandContext context)
 
         public async Task<CommandResult> Pokedex(CommandContext context)
         {
-            User user = (await context.ParseArgs<Optional<User>>()).OrElse(context.Message.User);
+            (Optional<User> optionalUser, Optional<string> mode_, Optional<User> optionalUser2) =
+                await context.ParseArgs<Optional<User>, Optional<string>,Optional<User>>();
+            User user = optionalUser.IsPresent ? optionalUser.Value : context.Message.User;
             bool isSelf = user == context.Message.User;
-            int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
-            return new CommandResult
+
+            // Have to additionally check if the string is not empty, otherwise "!dex " would jump right into the else - branch.
+            //   Not sure if this is an actual issue when connected to twitch
+            if( !mode_.IsPresent || string.IsNullOrWhiteSpace( mode_.ToString() ) )
             {
-                Response = isSelf
-                    ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
-                    : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
-            };
+                int numUniqueSpecies = (await _badgeRepo.CountByUserPerSpecies(user.Id)).Count;
+                return new CommandResult
+                {
+                    Response = isSelf
+                        ? $"You have collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                        : $"{user.Name} has collected {numUniqueSpecies} distinct Pokémon badge(s)"
+                };
+            }
+            else
```

Comment:
Since the if-case returns, you can omit the else-block and avoid the additional layer of indentation

Reply:
Makes sense. I removed every **else** since all of them were hiding behind return - statements